std_msgs/Byte number
std_msgs/Byte condition
std_msgs/Byte environment
geometry_msgs/Point context
std_msgs/Bool before_after
std_msgs/UInt32 number_of_updates

---
